test/extended/cli/builds.go

```diff
@@ -0,0 +1,179 @@
+package cli
+
+import (
+	"encoding/json"
+	"fmt"
+	g "github.com/onsi/ginkgo"
+	o "github.com/onsi/gomega"
+	exutil "github.com/openshift/origin/test/extended/util"
+	"k8s.io/kubectl/pkg/util/slice"
+)
+
+const (
+	bcOutputTemplate      = "{{with .spec.output.to}}{{.kind}} {{.name}}{{end}}"
+	testDockerFileContent = "FROM image-registry.openshift-image-registry.svc:5000/openshift/tests:latest"
+)
+
+type expectedBuild struct {
+	bcName      string
+	isName      string
+	sourceType  string
+	outputType  string
+	outputImage string
+}
+
+func (r *expectedBuild) setDefaults() {
+	if r.outputType == "ImageStreamTag" && r.isName == "" {
+		r.isName = r.bcName
+	}
+}
+
+func testNewBuild(oc *exutil.CLI, opts expectedBuild) {
+	opts.setDefaults()
```

Comment:
We can move this function to `g.JustBeforeEach` as a prerequisite step when each `g.By` becomes a `g.It`

Reply:
removed the config and simplified the flow so `g.JustBeforeEach` is not needed anymore